### PR TITLE
Added validations to New Patient Dialog

### LINKED ISSUE #31: Unexplained crash (bug report attached)
This happened; I don't know why.

Bug report: https://www.dropbox.com/s/osa648gwmei8be0/bugreport-2015-09-24-23-33-06.txt?dl=0

### PR DESCRIPTION
**Pull request by llvasconcellos**
Added validations to New Patient Dialog.
Renamed NewPatientActivityTest to EditPatientDialogFragmentTest.
This fixes #31

### CODE REVIEW

File: app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java [hunk 1]
```diff
@@ -174,18 +198,40 @@ public void onSubmit() {
...
         populateFields(args);
 
-        AlertDialog dialog = new AlertDialog.Builder(getActivity())
+        final AlertDialog dialog = new AlertDialog.Builder(getActivity())
             .setCancelable(false) // Disable auto-cancel.
             .setTitle(title)
-            .setPositiveButton(getResources().getString(R.string.ok), new DialogInterface.OnClickListener() {
+            .setPositiveButton(getResources().getString(R.string.ok), new DialogInterface
+                .OnClickListener() {
                 @Override public void onClick(DialogInterface dialogInterface, int i) {
```
Comment (G1DR4) [inline app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java:204]: is this override empty now?

File: app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java [hunk 2]
```diff
@@ -119,23 +120,46 @@ public void onSubmit() {
...
+                case R.id.patient_sex_female:
+                    sex = Patient.GENDER_FEMALE;
+                    break;
+                case R.id.patient_sex_male:
+                    sex = Patient.GENDER_MALE;
+                    break;
+            }
         }
+
+
```
Comment (viniciusboson) [inline app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java]: Here should be only one empty line.

File: app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java [hunk 3]
```diff
@@ -119,23 +120,46 @@ public void onSubmit() {
...
+        if (!valid) {
+            throw new ValidationException("Validation Error");
+        }
+
```
Comment (viniciusboson) [inline app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java:160]: Here has an extra empty line, as well.

**Comment by zestyping** [comment]
Let's hold off on adding patient validation for now.  My reasons for not doing this are:
- We do want to allow patients to have no specified gender, and to let clinicians go back and fill in the gender later by editing the patient.
- By embedding the limit of 120 years into our code, we create coupling between the internals of our client and the internals of OpenMRS.  The 120-year limit is undocumented behaviour of the OpenMRS API, so we shouldn't write our code to assume it.  Instead, we should handle error messages returned by OpenMRS gracefully, giving good feedback to the user if it's a problem that the user can correct.
- Even if we do want to do validation, it's clunky to check on submit and then fail with an error message; it's better to do validation live as the user types, and disable the submission button if the input is invalid. 
- Perhaps more controversially, I think OpenMRS should not be limiting the age to 120 years at all.  I think that it was a poor design decision to embed this restriction into their API, that 120 years is an arbitrary choice, and that it is not the storage layer's job to enforce business logic restrictions like that.  OpenMRS should trust its clients — let the clients apply or not apply restrictions as best suits their purpose, and let the clients provide the UI necessary to make those restrictions make sense to users.

More importantly, at the moment, this isn't a high enough priority right now to spend time on it.  We don't need it for the field trip.

**Comment by llvasconcellos** [comment]
@zestyping You are right!

**Pull request merged**